Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -648,12 +719,23 @@ namespace Terrain
     template<typename Callback>
     void TerrainMeshManager::ForOverlappingSectors(const AZ::Aabb& bounds, Callback callback)
     {
+        const AZ::Vector2 boundsMin2d = AZ::Vector2(bounds.GetMin().GetX(), bounds.GetMin().GetY());
+        const AZ::Vector2 boundsMax2d = AZ::Vector2(bounds.GetMax().GetX(), bounds.GetMax().GetY());
+
         for (uint32_t lodLevel = 0; lodLevel < m_sectorStack.size(); ++lodLevel)
         {
+            // Expand the bounds by the spacing of the lod since vertex normals are affected by neighbors.
+            const AZ::Vector2 lodSpacing = AZ::Vector2(m_sampleSpacing * (1 << lodLevel));
+            const AZ::Vector2 lodBoundsMin2d = boundsMin2d - lodSpacing;
+            const AZ::Vector2 lodBoundsMax2d = boundsMax2d + lodSpacing;
+
             auto& stackData = m_sectorStack.at(lodLevel);
             for (StackSectorData& sectorData : stackData.m_sectors)
             {
-                if (sectorData.m_aabb.Overlaps(bounds))
+                const AZ::Vector2 sectorAabbMin2D = AZ::Vector2(sectorData.m_aabb.GetMin().GetX(), sectorData.m_aabb.GetMin().GetY());
+                const AZ::Vector2 sectorAabbMax2D = AZ::Vector2(sectorData.m_aabb.GetMax().GetX(), sectorData.m_aabb.GetMax().GetY());
```

Comment:
Same here - can just feed the Vector3 into the Vector2 constructor.

Reply:
oh nice, didn't know vector2 had that constructor. I'll do that in the next PR though unless AR fails.